app/repositories/ContentAPI.scala

```diff
@@ -22,19 +22,43 @@ object ContentAPI {
   private val apiClient = new LiveContentApiClass(Config().capiKey, Config().capiUrl)
   private val previewApiClient = new DraftContentApiClass(Config().capiKey, Config().capiPreviewUrl)
 
-  def countOccurencesOfTagInContents(contentIds: List[String], apiTagId: String): Int= {
+  def countOccurencesOfTagInContents(contentIds: List[String], apiTagId: String): Int = {
+    val builder = StringBuilder.newBuilder
+    var pageSize = 0
+
+    var total = 0
+
```

Comment:
Not sure if it's just me but this is quite hard to read, I think it's the countTags in 2 places.

Could you use the for loop to construct an array of strings ready for the various requests that you'll make, then can you could use a fold, making requests and combining count.